src/animation.js

```diff
@@ -281,17 +278,19 @@ class AnimationPlayer {
           }
         }
       } catch (e) {
-        dev().error(TAG_, 'completion failed: ' + e, e);
-        success = false;
+        error = dev().createError('completion failed', e);
+        dev().error(TAG_, error);
       }
     }
-    if (success) {
-      this.resolve_();
+    if (error) {
+      this.reject_(dev().assertError(error));
```

Comment:
Yes, but it's really hard to get that in the linting system. Just using `#assertError` takes care of like 10 different `if (error)`, `if (error instanceof Error)`, etc, cases.